### DISCUSSION TITLE
Need help finding OPA Gatekeeper K8s support matrix

### DISCUSSION THREAD

**Discussion by samkulkarni20** [open-policy-agent/community#213] [Gatekeeper] [answered]
Hi All!
I'm trying to find the OPA Gatekeeper K8s support matrix. So far, I could only find this: https://open-policy-agent.github.io/gatekeeper/website/docs/install/#minimum-kubernetes-version. Although, this line is constant across documentation for all supported Gatekeeper versions viz. 3.6-3.8. Hence, IMO, this line is not a very reliable way of telling which all k8s versions are supported by a given Gatekeeper version.
e.g. K8s recently released v1.24 on May 4th, 2022, but Gatekeeper 3.6 is around since 31st Aug 2021. Now according to the documentation line above, 3.6 supports k8s v1.24, from the day it was released, but does it really? And even if that was true, can it be reliably considered true for all new versions?

Am I missing something? Is there a better documentation available for Gatekeeper support matrix somewhere?
TIA!

**Comment by sozercan** [comment, marked as answer]
Minimum version of K8s for Gatekeeper is 1.16 as per the doc. See the K8s version support doc here: https://github.com/open-policy-agent/gatekeeper/blob/master/docs/Release_Management.md#supported-kubernetes-versions

**Reply by sozercan** [reply]
added further clarification here: https://github.com/open-policy-agent/gatekeeper/blob/master/docs/Release_Management.md#supported-kubernetes-versions

in the above example, v3.6 is not a supported version of Gatekeeper so it is not assumed to be compatible with K8s v1.24. However, v3.7 is a supported version of Gatekeeper (at the time of this writing), so it is assumed to be compatible with K8s v1.24.

**Comment by KayenM** [comment]
Hi there, I am planning the integration of OPA Gatekeeper into a v1.24.z Kubernetes cluster and am in a similar boat where I am unable to find clear documentation on historical compatibility between Gatekeeper and Kubernetes versions. 

The [Gatekeeper release management docs](https://github.com/open-policy-agent/gatekeeper/blob/master/docs/Release_Management.md) provide the following info:

- Gatekeeper’s support policy is that they ensure compatibility of currently supported Gatekeeper versions with currently supported Kubernetes versions.
- Gatekeeper supports n-1 minor release versions where n is the current release. 

Thus I have assumed that at any point in time where there is overlap between a supported version of Gatekeeper and a supported version of Kubernetes, they will be compatible.

From this, I created the following compatibility matrix. I'd be really grateful if a member of the OPA Gatekeeper team could verify the matrix to ensure it is accurate. Thank you!

[Compatibility Matrix Spreadsheet](https://docs.google.com/spreadsheets/d/1uDjpVMEhTClunMU1j8rascHaK9LnpmwdvAMb6_ADkC4/edit?usp=sharing)

<img width="1455" alt="Screenshot 2024-10-23 at 7 13 28 PM" src="https://github.com/user-attachments/assets/864f793e-7e19-4693-a365-744ebd75e7cf">

**Reply by ritazh** [reply]
Thanks for creating this matrix! (would you like to contribute this to the project by adding it to the website?) For officially supported k8s and GK versions, it looks correct.

Note:
In terms of testing and back compat, we typically still test older EOL k8s versions until one of us makes an update to https://github.com/open-policy-agent/gatekeeper/blob/17be6a6eb68a8bfa99b2d15fdb54673130ea06a2/.github/workflows/workflow.yaml#L36

**Reply by KayenM** [reply]
Hi Rita, I appreciate your response, your link to the workflow build_test job is very helpful!

I'd be thrilled to contribute to the project, and will get a PR out soon (following the [contribution process](https://open-policy-agent.github.io/gatekeeper/website/docs/help/#contributing-process)). 😄

**Reply by KayenM** [reply]
Hey Rita, I've made the following PR to add this to the release management docs: https://github.com/open-policy-agent/gatekeeper/pull/3664.

Would appreciate any feedback!
